submariner/centralized-control-plane.md

```diff
@@ -0,0 +1,298 @@
+# Centralized Service Control Plane
+
+## Summary
+
+Current Submariner service management follows the model and guidelines
+ defined in [MCS](https://github.com/kubernetes-sigs/mcs-api). Specifically the
+ current design has these attributes:
+
+- Exported services are automatically reflected into all clusters in the ClusterSet;
+- Service are exported and imported using the same namespace and name; and
+- Service exports are managed from workload clusters.
+
+The above works well in an environment where clusters are used by a single
+ administrative domain and services are commonly shared across all clusters.
+ For example, when a company runs clusters as the runtime infrastructure and developers can
+ deploy to any cluster for availability, redundancy, or geographic proximity. However, it is
+ less suited in a deployment model where clusters are owned and operated by different teams,
+ each having independent control and configuration (see next section for elaboration).
```

Comment:
One has to ask themself, is MCS API and Submariner even suitable for such an environment?

MCS and Submariner are rather aligned on simplicity of operations for the user, and hence the "clusterset" model is a simple world where the aforementioned principles apply.

If the desire is simply to share *some* services between clusters that are not expected to be operated by the same people and have a "flat network" type of trust model, perhaps we shouldn't be applying MCS/Submariner at all since a better approach might be exposing said services via `Ingress` services?

Reply:
the MCS API is relatively abstract and not opinionated about how to implement the solution. We can still use it to program the Submariner data plane. The proposal is a superset of the functionality currently provided (i.e., publish to all without name changes - proposal is extending MCS)
Ingress is a less optimal as it becomes the destination/source of traffic instead of the originators (unless we start doing SSL without termination or passthrough....).